src/utils.js

```diff
@@ -112,3 +60,84 @@ export function buildClause (field, value, opts) {
 
   return Object.assign({}, mainClause, opts)
 }
+
+export function toBool (filters) {
+  const unwrapped = {
+    must: unwrap(filters.and),
+    should: unwrap(filters.or),
+    must_not: unwrap(filters.not),
+    minimum_should_match: filters.minimum_should_match
+  }
+
+  if (
+    filters.and.length === 1 &&
+    !unwrapped.should &&
+    !unwrapped.must_not
+  ) {
+    return unwrapped.must
+  }
+
+  const cleaned = {}
+
+  if (unwrapped.must) {
+    cleaned.must = unwrapped.must
+  }
+  if (unwrapped.should) {
+    cleaned.should = filters.or
+  }
+  if (unwrapped.must_not) {
+    cleaned.must_not = filters.not
+  }
+  if (
+    unwrapped.minimum_should_match &&
+    unwrapped.minimum_should_match <= filters.or.length
```

Comment:
The issue is that the feature was included in 2.1, so I didn't want to remove it to avoid a breaking change. Although previously it only checked that there are at least 2 `should` arguments and didn't worry about the number passed to `minimum_should_match`